mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodEventHandler.m

```diff
@@ -296,8 +301,12 @@ - (BOOL) handleKeymanEngineActions:(NSEvent *)event in:(id) sender {
     BOOL deleteBackPosted = NO;
     NSArray *actions = nil;
     if (![self willDeleteNullChar]) {
-        NSEvent *eventWithOriginalModifierFlags = [NSEvent keyEventWithType:event.type location:event.locationInWindow modifierFlags:self.AppDelegate.currentModifierFlags timestamp:event.timestamp windowNumber:event.windowNumber context:event.context characters:event.characters charactersIgnoringModifiers:event.charactersIgnoringModifiers isARepeat:event.isARepeat keyCode:event.keyCode];
-        actions = [self.kme processEvent:eventWithOriginalModifierFlags];
+        if (self.AppDelegate.lowLevelEventTap != nil) {
+            NSEvent *eventWithOriginalModifierFlags = [NSEvent keyEventWithType:event.type location:event.locationInWindow modifierFlags:self.AppDelegate.currentModifierFlags timestamp:event.timestamp windowNumber:event.windowNumber context:event.context characters:event.characters charactersIgnoringModifiers:event.charactersIgnoringModifiers isARepeat:event.isARepeat keyCode:event.keyCode];
+            actions = [self.kme processEvent:eventWithOriginalModifierFlags];
+        }
+        else // this probably isn't going to work so well, but as a fallback it's hopefully better than nothing.
+            actions = [self.kme processEvent:event];
```

Comment:
Brace yourself :)

Reply:
FWIW, This is what the code used to do before I added the event tap code. Depending on the client app and the keyboard, most things should work okay. It was originally added to support chirality, and later enhanced to detect command-key presses so we could re-get the context. I'll update the comment to be more informative.